_releases/2023-01-09-1.7.0-released.md

```diff
@@ -0,0 +1,109 @@
+may lead to the compiler rejecting a regex literal for syntax errors, even if it's valid for the runtime engine.
+Read more about the implications on the language syntax in [#12857](https://github.com/crystal-lang/crystal/issues/12857).
+
+# Misc
+
+## Language
+
+* Lib functions are now visible in top-level macros ([#12848](https://github.com/crystal-lang/crystal/pull/12848)).
+* Added `ArrayLiteral#-(other)` and `TupleLiteral#-(other)` ([#12646](https://github.com/crystal-lang/crystal/pull/12646)).
+* `annotation` supports the `@[Deprecated]` annotation now ([#12557](https://github.com/crystal-lang/crystal/pull/12557)).
+* Removed oct/bin floating point literals. This number syntax had been broken and isn't very useful anyway... ([#12687](https://github.com/crystal-lang/crystal/pull/12687)).
+
+## Compiler
+
+Clear error message when declaring a constant within another constant declaration ([#12566](https://github.com/crystal-lang/crystal/pull/12566)).
+
+```cr
+A = B = 1 # Syntax Error: dynamic constant assignment
+```
+
+Fixed wrong/missing locations of various AST nodes in the parser ([#11798](https://github.com/crystal-lang/crystal/pull/11798)).
+
+⚠️ Obsolete FreeBSD 11 compatibility was dropped ([#12612](https://github.com/crystal-lang/crystal/pull/12612)).
+
+## stdlib
+
+⚠️ `File.real_path` was renamed to `File.realpath`. This spelling just makes more sense as it's used everywhere else. The old version is deprecated, but keeps working until the next major release ([#12552](https://github.com/crystal-lang/crystal/pull/12552)).
+
+⚠️ Error handling for `libXML2` does no longer leak errors outside the current context. `XML.errors` has been deprecated, but should continue to work. We highly recommend to access errors directly in the respective context via `XML::Reader#errors` and `XML::Node#errors` ([#12663](https://github.com/crystal-lang/crystal/pull/12663), [#12795](https://github.com/crystal-lang/crystal/pull/12795)).
+
+`File#flock_shared` and `File#flock_exclusive` are now fiber-aware and when they block, they're no longer blocking the thread ([#12861](https://github.com/crystal-lang/crystal/pull/12861)).
+They're now also supported on Windows ([#12766](https://github.com/crystal-lang/crystal/pull/12766)).
+
+`BigDecimal#to_s` supports scientific notation ([#10805](https://github.com/crystal-lang/crystal/pull/10805)).
+
+```crystal
+BigDecimal.new(Float32::MAX) # => 3.4028235e+38
+```
+
+Further additions:
+
+* Automatically cast Int to Float for `{JSON,YAML}::Any#as_f` ([#12835](https://github.com/crystal-lang/crystal/pull/12835)).
+* Several missing `#index!`/`#rindex!` methods on `String` and `Indexable`([#12730](https://github.com/crystal-lang/crystal/pull/12730), [#12759](https://github.com/crystal-lang/crystal/pull/12759)).
+* Methods to manipulate `SemanticVersion` ([#12834](https://github.com/crystal-lang/crystal/pull/12834).
```

Comment:
```suggestion
* Methods to manipulate `SemanticVersion` ([#12834](https://github.com/crystal-lang/crystal/pull/12834), see also the [docs](https://crystal-lang.org/api/1.7.0/SemanticVersion.html)).
```

Reply:
Yeah, I haven't added all the docs links yet. I was planning to link the class name directly. Same for all the other features mentioned in the post.